R/02-filter.R

```diff
@@ -18,18 +23,57 @@
+#' gimap_dataset <- gimap_filter(gimap_dataset, filter_type = "zero_count_only") 
+#' #or 
+#' gimap_dataset <- gimap_filter(gimap_dataset, filter_type = "low_plasmid_cpm_only")
+#' 
+#' # 
 #'
 #' }
 #'
 
 gimap_filter <- function(.data = NULL,
                          gimap_dataset,
-                         filter_type = "both") {
+                         filter_type = "both",
+                         cutoff = NULL,
+                         filter_zerocount_target_col = NULL,
+                         filter_plasmid_target_col = NULL,
+                         min_n_filters = 1) {
 
   if (!is.null(.data)) gimap_dataset <- .data
 
   if (!("gimap_dataset" %in% class(gimap_dataset))) stop("This function only works with gimap_dataset objects which can be made with the setup_data() function.")
-
+  
+  #check filter type input to make sure that it is a supportable input
+  if (!(filter_type %in% c("both", "zero_count_only", "low_plasmid_cpm_only"))) stop("Specification for `filter_type` not understood; Need to use 'both', 'zero_count_only', or 'low_plasmid_cpm_only'")
+  
+  zc_filter <- NULL
+  p_filter <- NULL
+  #*ADD any new filters here* assigning it a NULL value
+  
+  #This section calls the appropriate filtering functions and assigns results to the filter variables assigned NULL earlier (they will stay NULL if there filter wasn't selected to be run according to the input to the function)
+  if (filter_type == "both"){
+    zc_filter <- qc_filter_zerocounts(gimap_dataset, filter_zerocount_target_col = filter_zerocount_target_col)$filter
+    p_filter <- qc_filter_plasmid(gimap_dataset, cutoff = cutoff, filter_plasmid_target_col = filter_plasmid_target_col)$plasmid_filter
+  } else if (filter_type == "zero_count_only"){
+    zc_filter <- qc_filter_zerocounts(gimap_dataset, filter_zerocount_target_col = filter_zerocount_target_col)$filter
+  } else if(filter_type == "low_plasmid_cpm_only"){
+    p_filter <- qc_filter_plasmid(gimap_dataset, cutoff = cutoff, filter_plasmid_target_col = filter_plasmid_target_col)$plasmid_filter
+  }
+  
+  
+  possible_filters <- list(zc_filter, p_filter)
+  #*ADD any new filters here* within the list of `possible_filters`
+  
+  #this first cbinds each filter enumerated in possible_filters together (no matter how many there are, and ignores the NULLs) using the reduce function
+  #then it finds the row sum (how many are filters flagged each construct e.g., number of TRUE in each row), 
+  #and finally compares the row sum to the `min_n_filters` parameter to report TRUEs and FALSEs according to whether each construct is flagged by the minimum number of required filters
+  #TRUE means it should be filtered, FALSE means it shouldn't be filtered
+  combined_filter <- rowSums(reduce(possible_filters, cbind)) >= min_n_filters 
+  
+  
   gimap_dataset$filtered <- NULL #TODO: Filtered version of the data can be stored here
```

Comment:
Add here a nice readable version of combined_filter to `gimap_dataset` and then the filtered version of the data to `filtered_data`

Reply:
This is in a later PR so I think this means this is set to go!